File: beecrowd/c++/beginner/1013-the-greatest.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  int s[3], max;

  cin >> s[0] >> s[1] >> s[2];

  for (int i = 0; i < 3; i++) {
    if (s[i] > max) {
      max = s[i];
    }
  }

  cout << max << " eh o maior" << endl;

  return 0;
}

File: beecrowd/dart-2.9/beginner/1013-the-greatest.dart
import 'dart:convert';
import 'dart:io';
import 'dart:math' as math;

void main() {
  final inputs = _getIntListLine();

  print('${_calcMax(inputs)} eh o maior');
}

int _calcMax(List<int> inputs, [int max, int i = 0]) {
  max = math.max(max ?? inputs[i], inputs[++i]);

  if (i < inputs.length - 1) {
    return _calcMax(inputs, max, i);
  }

  return max;
}

String _getLine() {
  final line = stdin.readLineSync(encoding: Encoding.getByName('utf-8'));

  return line.trim();
}

List<int> _getIntListLine() => _getLine().split(' ').map(int.parse).toList();
